### ISSUE TITLE
Class generation issues

### ISSUE THREAD

**Issue by inumanag** [seq-lang/seq#125]
This code segfaults:
```
class Bloom:
     filter: list[int]
     top: dict[K, int]
     def __init__(self: Bloom, m: int):
         self.filter = [0 for _ in range(m)]
         # self.top = dict[K, int]() # segfaults 
```

When the last line is not commented, it works. Can we automatically call default constructors for each class member in `__new__`? It is really hard to find these errors with classes that have many members, especially since it is not yet possible to debug Seq code.

Also, can we have an attribute (say, `@compare`) for automatic generation of comparison functions (`__lt__`, `__gt__` etc); maybe pickling as well? We can reuse these functions from class members? Python has something similar for dataclasses.

**Comment by arshajii** [comment]
Where does `Bloom` fail? I think it makes sense to set things to `None` by default rather than call constructors (since that might have unintended side effects), and then have good support for `None` in debug mode.

Can we generate `__lt__` etc. in the frontend if there is a method tagged `@compare`? e.g.

```python
class Foo:
    # user writes this method
    @compare
    def cmp(self: Foo, other: Foo) -> int:
        return ...

    # these methods get automatically added by the front-end / parser
    # name 'cmp' and type of 'other' copied from tagged method
    def __lt__(self: Foo, other: Foo) -> bool:
        return self.cmp(other) < 0
    # same for gt, le, ge

We could probably generate it in the backend but this seems like a relatively straightforward transformation that might make more sense in the frontend.

For picking, pickle-able members should all be taken care of automatically, so I've found that 95% of the time I don't need to define any new methods for it to work. Did you have a particular case that didn't work automatically?

**Comment by inumanag** [comment]
Resolved in #132
